src/Handler/User.hs

```diff
@@ -37,28 +37,17 @@ getUsersR :: Handler Html
 getUsersR = do
     void requireAuth
 
-    users' <- runDB $
-                  select $
-                  from $ \user -> do
-                  orderBy [desc $ user ^. UserId]
-                  return user
-
-    infos :: [(Entity User, ((Value Text, Value Text), Value Role))] <- runDB $
-        select $
-        from $ \(user `InnerJoin` role `InnerJoin` project) -> do
-        on_ (project ^. ProjectId ==. role ^. ProjectUserRoleProject)
-        on_ (user ^. UserId ==. role ^. ProjectUserRoleUser)
-        return (user, ((project ^. ProjectName, project ^. ProjectHandle), role ^. ProjectUserRoleRole))
-
-
-    let users = map (\u -> (getUserKey u :: Text, u)) $ filter isVisible users'
-        infos' :: [(UserId, ((Text, Text), Role))] = map (entityKey *** unwrapValues) infos
-        infos'' :: [(UserId, Map (Text, Text) (Set Role))] = map (second $ uncurry M.singleton . second S.singleton) infos'
-        allProjects :: Map UserId (Map (Text, Text) (Set Role)) = M.fromListWith (M.unionWith S.union) infos''
+    (users', allProjects) <- runDB (liftA2 (,) fetchAllUsersDB fetchAllUserProjectInfosDB)
```

Comment:
IMHO, both are equally readable to the professional Haskeller. The only thing that is actually better is (applicative-)do notation, and that for the newcomer.